src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -340,7 +340,12 @@ public sealed class UIActionEngine
             return await Task.FromResult(true);
 #else
             var newProjectNames = new List<string>();
-            var msBuildProjects = uiService.Projects.Where(project => project is MSBuildNuGetProject);
+
+            // check if project is packages.config and it's dte project instance can also be converted to VSProject4.
+            // otherwise don't show format selector dialog for this project
+            var msBuildProjects = uiService.Projects.Where(project =>
+                project is MSBuildNuGetProject &&
+                (project as MSBuildNuGetProject).MSBuildNuGetProjectSystem.VSProject4 != null);
```

Comment:
`You could just try casting MSBuildNuGetProjectSystem to a VSMSBuildNuGetProjectSystem `
I can't do that since VSMSBuildNuGetProjectSystem isn't available inside NuGet.PackageManagement.UI project. 

I thought about bool property but then realized what if we need to access VSProject4 at later stage similar to EnvDteProject so didn't see any harm getting VSProject4 and validating.